### ISSUE TITLE
Config dependencies

### ISSUE THREAD

**Issue by Tarik02** [nuxt/nuxt.js#3610]
### What problem does this feature solve?

Sometimes nuxt config contains not only of `nuxt.config.js`, but other files. For example i use dotenv:
```js
const dotenv = require("dotenv");

const env = dotenv.config().parsed;

module.exports = {
  env,
```
But there's a problem. Nuxt watches only for `nuxt.config.js` and there's no ability to add files to watch for.

### What does the proposed changes look like?

Something like `dependencies` in `nuxt.config.js`:
```js
module.exports = {
  dependencies: [".env"],
```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c7465">#c7465</a>)</em></sub></div>

**Comment by manniL** [comment]
You can already add file that should be watch with the [`build.watch` property](https://nuxtjs.org/api/configuration-build/#watch)

**Comment by Tarik02** [comment]
This does not help. `watch` does not reload `nuxt.config.js`

**Comment by Atinux** [comment]
Related to https://github.com/nuxt/nuxt.js/pull/3633

**Comment by manniL** [comment]
@galvez Do you think this has been resolved through #3633? :relaxed:

**Comment by Tarik02** [comment]
Resolved with #3633

**Comment by galvez** [comment]
@manniL @Tarik02 please note it's the `watch` property, not `build.watch` ;)
